packages/core/src/browser/theming.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import { CommandRegistry, CommandContribution, CommandHandler, Command } from '../common/command';
+import { QuickOpenService } from './quick-open/quick-open-service';
+import { QuickOpenModel, QuickOpenItem, QuickOpenMode } from './quick-open/quick-open-model';
+import { Emitter } from '../common/event';
+
+const dark = require('../../src/browser/style/variables-dark.useable.css');
+const light = require('../../src/browser/style/variables-bright.useable.css');
+
+export interface Theme {
+    id: string;
+    label: string;
+    description?: string;
+    editorTheme?: string;
+    activate(): void;
+    deactivate(): void;
+}
+
+export interface ThemeChangeEvent {
+    newTheme: Theme;
+    oldTheme?: Theme;
+}
+
+const darkTheme: Theme = {
+    id: 'dark',
+    label: 'Dark Theme',
+    description: 'Bright fonts on dark backgrounds.',
+    editorTheme: 'vs-dark',
+    activate() {
+        dark.use();
+    },
+    deactivate() {
+        dark.unuse();
+    }
+};
+
+const lightTheme: Theme = {
+    id: 'light',
+    label: 'Light Theme',
+    description: 'Dark fonts on light backgrounds.',
+    editorTheme: 'vs',
+    activate() {
+        light.use();
+    },
+    deactivate() {
+        light.unuse();
+    }
+};
+
+export class ThemeService {
```

Comment:
Could it be provided via the DI? And also `darkTheme` and `lightTheme` configured in the DI context?

Reply:
no, it is too late. we want to set the theme before everything.